### PR TITLE
Add basic LFSC utilities

### PR DESCRIPTION
**Pull request by ajreynol**
Adds basic utilities in preparation for the LFSC proof conversion.

Depends on https://github.com/cvc5/cvc5/pull/6881, review that first.

### CODE REVIEW

File: src/proof/lfsc/lfsc_util.h [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+ * ****************************************************************************
+ *
+ * Utilities for LFSC proofs.
+ */
+
+#include "cvc5_private.h"
+
+#ifndef CVC4__PROOF__LFSC__LFSC_UTIL_H
+#define CVC4__PROOF__LFSC__LFSC_UTIL_H
+
```
Comment (HanielB) [inline src/proof/lfsc/lfsc_util.h:20]: CVC5

File: src/proof/lfsc/lfsc_util.h [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  LAMBDA,
+  // a proof-let "plet"
+  PLET,
+  //----------- unknown
+  UNKNOWN,
+};
+
+/**
+ * Converts a lfsc rule to a string. Note: This function is also used in
+ * `safe_print()`. Changing this function name or signature will result in
```
Comment (HanielB) [inline src/proof/lfsc/lfsc_util.h:71]: say where this method is

**Review by HanielB — CHANGES_REQUESTED** [review]

File: src/proof/lfsc/lfsc_util.h (hunk 2 above)
Comment (ajreynol) [reply src/proof/lfsc/lfsc_util.h:71]: Which method? Note this is the standard documentation for all methods of this form.

Comment (HanielB) [reply src/proof/lfsc/lfsc_util.h:71]: `safe_print()`, I meant. But ok, I had never noticed the methods of this form nor knew about this safe_print stuff.

**Review by HanielB — APPROVED** [review]

**Pull request merged**